client/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,51 @@
+import { h } from "preact";
+import cx from "classnames";
+
+import styles from "./sidebar.css";
+
+const BASE_CLASS = "sidebar";
+
+const Sidebar = ({ data, handleItemChange, activeIndex }) => {
+  // Get the length of the matching function
+  const getFirstFunctionMatchLength = list => {
+    if (!list.length) {
+      return null;
+    }
+    if (!list[0] || !list[0].matchingCode) {
```

Comment:
Don't assume safety check. It's better to fail and fix the bug rather than work around for not having a bug.